src/kv/tx_pimpl.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/tx.h"
+
+namespace kv
+{
+  struct BaseTx::PrivateImpl
```

Comment:
Now that it's all done, it's clear that this pimpl pattern may be unnecessary; we already have an abstract `Tx`, implemented by a concrete `CommittableTx`, which could hold these fields? But that's easy enough to change to, and has the same effect of breaking the public dependency.

Reply:
Had a look at this while restoring the move-assignment operator, thinking maybe only `CommittableTx` needs to be move-assignable? But it turns out we do instantiate other types in the chain, notably `ReadOnlyTx`. This needs to have all of the same handle-management logic, so we need to implement these in `BaseTx` (or thereabouts), so a PrivateImpl makes sense.